### PR TITLE
fix(server): refactor tool_result parsing and harden transforms

### LINKED ISSUE #601: refactor(server): extract shared tool_result parsing from cli-session and sdk-session
## Context

PR #596 adds identical `case 'user':` handlers in both `cli-session.js` and `sdk-session.js` for extracting `tool_result` content blocks from user-role messages during the tool loop. The logic is duplicated:

- Parse `block.content` (string or array of text blocks)
- Truncate at `MAX_TOOL_RESULT_SIZE` (10KB)
- Emit `tool_result` event with `{ toolUseId, result, truncated }`

## Proposal

Extract the shared parsing into a helper function (e.g. in a new `tool-result.js` or in an existing shared module) that both session classes can call.

This keeps the two session implementations in sync and reduces the surface area for divergence if the result format changes.

## Files

- `packages/server/src/cli-session.js` — lines ~476-506
- `packages/server/src/sdk-session.js` — lines ~252-282

### LINKED ISSUE #602: voiceCleanup: strip mid-sentence filler words
**From:** PR #595 review (message transform pipeline)

The `voiceCleanup` transform only strips filler words at the very start of the message (`^(um|uh|like|so|okay|well|basically)`). Voice transcription commonly produces:
- Mid-sentence fillers: "please um fix the bug"
- Repeated fillers: "um, um, fix it"
- Post-comma fillers: "fix the bug, uh, in the server"

Consider using a global regex or multi-pass approach to handle these cases. Care needed to avoid false positives with legitimate uses of "like" ("I'd like to...") and "so" ("do so immediately").

### LINKED ISSUE #603: contextAnnotation: skip for short plan-mode responses
**From:** PR #595 review (message transform pipeline)

The `contextAnnotation` transform always prepends `[cwd: /foo, model: opus, ...]` to every message, including very short plan approval responses like "yes", "looks good", or "approve". This wastes context window tokens and may confuse the ExitPlanMode flow.

Consider:
- Skipping annotation for messages shorter than ~10 characters
- Adding a `skipAnnotation` flag in the transform context when in plan mode
- Only annotating the first message in a conversation (context doesn't change mid-session)

### PR DESCRIPTION
**Pull request by blamechris**
## Summary

- Extract shared `emitToolResults()` helper into `tool-result.js`, eliminating duplicate tool_result parsing logic in both `cli-session.js` and `sdk-session.js`
- Improve `voiceCleanup` transform to strip mid-sentence filler words ("um", "uh") after commas, before commas, and standalone
- Fix trailing comma artifact when filler word appears at end of sentence
- Skip `contextAnnotation` for short messages (< 10 chars) like plan approvals ("yes", "approve") to save context window tokens

Closes #601
Closes #602
Closes #603

Note: #604 (integration tests for session sendMessage transform wiring) remains open — requires subprocess/SDK mocking infrastructure not yet available.

## Test Plan

- [x] Server tests pass (`cd packages/server && PATH="/opt/homebrew/opt/node@22/bin:$PATH" npm test`) — 771/772 pass, 1 pre-existing flaky test (session-context, fixed in #630)
- [x] New `tool-result.test.js` with 7 test cases (string/array content, truncation, edge cases)
- [x] Updated `message-transform.test.js` with tests for mid-sentence fillers, trailing comma, filler-before-comma, short message skip, threshold length
- [x] Total: 26 new/updated tests pass

### CODE REVIEW

File: packages/server/tests/message-transform.test.js [hunk 1]
```diff
@@ -74,6 +89,21 @@ describe('voiceCleanup transform', () => {
...
+
+  it('removes standalone um/uh mid-sentence', () => {
+    const result = transform('please um fix the bug', { isVoiceInput: true })
+    assert.equal(result, 'please fix the bug.')
+  })
+
+  it('removes multiple mid-sentence fillers', () => {
+    const result = transform('fix, uh, the um bug', { isVoiceInput: true })
+    assert.equal(result, 'fix, the bug.')
+  })
```
[the remark this answers is not in the document]
Comment (blamechris) [reply packages/server/tests/message-transform.test.js]: **FOLLOW-UP ISSUE**

This is tracked by #604 which this PR closes. Issue #604 specifically requests integration tests for session sendMessage transform wiring. However, a full integration test requires mocking the Claude subprocess/SDK `query()` call to intercept the transformed message — which is non-trivial and better done as a focused follow-up.

Reopening #604 to track the integration test gap separately from the unit test coverage added here.
